flocker/provision/_aws.py

```diff
@@ -60,14 +60,16 @@ def _wait_until_running(instance):
     with start_action(
         action_type=u"flocker:provision:aws:wait_until_running",
         instance_id=instance.id,
-    ):
-        while instance.state != 'running':
+    ) as context:
+        while instance.state != u'running' and instance.state != u'terminated':
```

Comment:
I agree.  And given that we poll for the state changes the picture is even more complicated for us because we can miss some intermediate transitions. So, for now I decided to wait until an instance is out of `pending` or `rebooting` state (`wait_until_running` is used with `reboot()` as well) and then check if the new state is actually `running`.
Also, I am adding logging of `instance.state_reason` field. I think that it should be either `state-reason-code`  or `state-reason-message`.